protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {IAtomicBridgeInitiator} from "./IAtomicBridgeInitiator.sol";
+import {IWETH10} from "./WETH/interfaces/IWETH10.sol";
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+
+contract AtomicBridgeInitiator is IAtomicBridgeInitiator, Initializable {
+    struct BridgeTransfer {
+        uint256 amount;
+        address originator;
+        bytes32 recipient;
+        bytes32 hashLock;
+        uint256 timeLock;
+        bool completed;
+    }
+
+    mapping(bytes32 => BridgeTransfer) public bridgeTransfers;
+    IWETH10 public weth;
+    address public authority;
+
+    modifier onlyAuthorized() {
```

Comment:
Additionaly, We are never checking `authority` at any point in the contract, so it seems redunant and will add gas. I'll remove, unless you can think of any maintance, or functionality uses that only an authorized wallet should be ablle to do.

Reply:
It was being used in `completeBridgeTansfer`, but as discussed in spain, we want the function to be callable by anyone. And this would allow the service to be decentralized with incentives later down the line. 

I've added Open Zeppelin Access Control and used it to only restrict the `refundBridgeTransfer` call. This function would only be called if for some reason we needed to refund funds.